app/objects/c_operation.py

```diff
@@ -349,33 +349,32 @@ async def _get_all_possible_abilities_by_agent(self, data_svc):
         return {a.paw: abilities for a in self.agents}
 
     def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors, agent_ran):
-        variables = re.findall(r'#{(.*?)}', self.decode_bytes(ability.test), flags=re.DOTALL) if ability.test else []
         if ability.ability_id in agent_ran:
             return
-        elif not agent.trusted:
+
+        valid_executors = ability.find_executors(agent.platform, agent_executors)
+
+        any_facts_fulfilled = False
+        for executor in valid_executors:
+            facts = re.findall(r'#{(.*?)}', executor.test, flags=re.DOTALL) if executor.command else []
+            if all(fact in op_facts for fact in facts):
+                any_facts_fulfilled = True
```

Comment:
1) Nice generator you got there
3) The `any_facts_fulfilled` vs. the `all(...)` call looks like a bug to me. Should any/all be aligned?
4) `all()` returns `True` if the input is empty--is that alright?

Reply:
Changed to `fact_dependency_fulfilled`: should be true if no facts are required or all facts are fulfilled